pkgs/applications/misc/gopass-ui/default.nix

```diff
@@ -0,0 +1,221 @@
+{ lib
+, system
+, stdenv
+, fetchgit
+, fetchurl
+, fetchFromGitHub
+, makeDesktopItem
+, makeWrapper
+, runCommand
+, writeTextFile
+, electron_7
+, electron-chromedriver_3 ? null
+, gifsicle
+, jq
+, libwebp
+, mozjpeg
+, nix-gitignore
+, nodejs-12_x
+, nodePackages
+, optipng
+, p7zip
+, pngquant
+, python2
+, utillinux
+, darwin
+, xcbuild
+, bash
+, symlinkJoin
+} @ args:
+
+let
+  electron = electron_7;
+
+  # Optimize the final space taken by our derivation
+  # by feeding electron builder with a tree of symlink
+  # instead of real file.
+  # It seem to effectively copy these symlink whenever no
+  # changes are made to the file.
+  symlinkedElectron = symlinkJoin {
+    name = "symlinked-electron";
+    paths = [ electron ];
+  };
+
+  # We found no sensible way to control electron-builder's
+  # unpacked dir output location. We thus need way to
+  # match what it will be on a per platform basis.
+  electronBuilderUnpackedDirname =
+      let
+        platform = stdenv.hostPlatform.system;
+        p2dn = {
+          i686-linux = "linux-ia32-unpacked";
+          x86_64-linux = "linux-unpacked";
+          armv7l-linux = "linux-armv7l-unpacked";
+          aarch64-linux = "linux-arm64-unpacked";
+          x86_64-darwin = "mac";
+        };
+      in
+    assert p2dn ? "${platform}";
+    p2dn."${platform}";
+in
+
+stdenv.mkDerivation rec {
+  pname = "gopass-ui";
+  version = "0.7.0";
+  name = "${pname}-${version}";
+
+  nativeBuildInputs = [
+    makeWrapper
+    nodejs-12_x
+  ];
+
+  src = fetchFromGitHub {
+    owner  = "codecentric";
+    repo   = "gopass-ui";
+    rev    = "v${version}";
+    sha256 = "1yn8x3raj8hqc6jshqdbn6yfkh6qmp2j5i6b3xz8q02xyvb1r241";
+  };
+
+  nodeDeps = (import ./node-dependencies.nix {
+      inherit system;
+      pkgs = args // { inherit electron-chromedriver_3; };
+      gopass-ui-src = src;
+    }).package;
+
+  buildInputs = [
+    electron
+  ] ++ stdenv.lib.optionals stdenv.isDarwin [ xcbuild darwin.DarwinTools ];
+
+  buildPhase = ''
+    dev_node_modules="${nodeDeps}/lib/node_modules/${pname}/node_modules"
+    ln -s -t "." "$dev_node_modules"
+
+    export PATH="$dev_node_modules/.bin:$PATH"
+
+    # Expansion of 'npm run build'.
+    NODE_ENV=production webpack --config webpack.main.prod.config.js
+    NODE_ENV=production webpack --config webpack.renderer.explorer.prod.config.js
+    NODE_ENV=production webpack --config webpack.renderer.search.prod.config.js
+  '';
+
+  doCheck = true;
+
+  checkPhase = ''
+    # Expansion of 'npm run release:check'.
+    tslint '{src,test,mocks}/**/*.{ts,tsx,js,jsx}' --project ./tsconfig.json
+    jest --testRegex '\.test\.tsx?$'
+  '';
+
+  installPhase = ''
+    ${if stdenv.isDarwin then ''
+      declare out_elec_bundle_dir="$out/Applications"
+      declare out_elec_exe="$out_elec_bundle_dir/Gopass UI.app/Contents/MacOS/Gopass UI"
+      declare out_elec_asar="$out_elec_bundle_dir/Gopass UI/Contents/Resources/app.asar"
```

Comment:
Unquoted path to .asar contains whitespace:
```sh
$ cat result/Applications/Gopass\ UI.app/Contents/MacOS/Gopass\ UI
#! /nix/store/my66alsy3dhj9iz9s3sq7c9sni1b1a2d-bash-4.4-p23/bin/bash -e
exec -a "$0" "/nix/store/ifx4mjs31j6bbkm5z7agn1li1gmrjh9f-gopass-ui-0.7.0/Applications/Gopass UI.app/Contents/MacOS/.Gopass UI-wrapped"  /nix/store/ifx4mjs31j6bbkm5z7agn1li1gmrjh9f-gopass-ui-0.7.0/Applications/Gopass UI/Contents/Resources/app.asar "$@"
```

Reply:
Argh. Indeed, you're right. Fixed this in b70e78d9f4a4116eb6a912bed7e23670c66db3a1.